lib/Dialect/Synth/Analysis/LongestPathAnalysis.cpp

```diff
@@ -730,7 +730,8 @@ class LocalVisitor {
   std::unique_ptr<llvm::ImmutableListFactory<DebugPoint>> debugPointFactory;
 
   // A map from the value point to the longest paths.
-  DenseMap<std::pair<Value, size_t>, SmallVector<OpenPath>> cachedResults;
+  DenseMap<std::pair<Value, size_t>, std::unique_ptr<SmallVector<OpenPath>>>
+      cachedResults;
```

Comment:
Unrelated, consider a `struct` over `std::pair` in a follow-on. This has the benefit of readability as well as possibly getting this onto one line. 😉